pandas/core/frame.py

```diff
@@ -8825,11 +8876,11 @@ def func(values: np.ndarray):
             # We only use this in the case that operates on self.values
             return op(values, axis=axis, skipna=skipna, **kwds)
 
-        def blk_func(values):
+        def blk_func(values, axis=1):
             if isinstance(values, ExtensionArray):
                 return values._reduce(name, skipna=skipna, **kwds)
             else:
-                return op(values, axis=1, skipna=skipna, **kwds)
+                return op(values, axis=axis, skipna=skipna, **kwds)
```

Comment:
this stays 1 for BlockManager but becomes 0 for ArrayManager?

Reply:
Yes, the default stays the same. And for BlockManager, this default is used. But this way, it allows us to override it for the ArrayManager reduce method.

Reply:
does axis ever actually get passed to blk_func?

Reply:
> does axis ever actually get passed to blk_func?

Yes, as mentioned above, the ArrayManager.reduce method uses this